### ISSUE TITLE
Insufficient color contrast between columns and error bar marker in dark mode

### ISSUE THREAD

**Issue by jperals** [highcharts/highcharts#23189]
#### Expected behaviour

There is enough color contrast between columns and their error bar markers, both in light and dark mode.

#### Actual behaviour

According to [the WebAIM color contrast checker](https://webaim.org/resources/contrastchecker/), 7 out of the 10 [default colors](https://api.highcharts.com/highcharts/colors) do not have enough color contrast with white, which is the color used for the error range in dark mode (attaching screenshot for the first color of the default palette, as an example).

Actually this also means that the default colors don't have enough contrast with the background in light mode, but we don't have this problem with the palette we use. However working with error bars is a more difficult problem because it is especially challenging to find a color for the error bar marker that will have enough contrast against both the column and the background.

One way to mitigate this would be to let the error bar marker have an outline or a semitransparent overlay, so that  either this outline or overlay or the inner stroke would have good contrast with any surrounding element.

#### Live demo with steps to reproduce
Fiddle: https://jsfiddle.net/canjhmyw/12/

#### Product version
Highcharts Core, v12

#### Affected browser(s)
Tested on Chrome

<img width="718" alt="Image" src="https://github.com/user-attachments/assets/1d16cc6f-72df-4428-a412-307c3d23cfe9" />

**Comment by KacperMadej** [comment]
Hi @jperals 

Thank you for submitting the feature request.

The colors are adjustable so you can change them to ones that match your needed contrast ratio.

With level AAA (ration 7:1) there is no color available to fit between black and white, so you will need a different color for one of modes (dark or light).

![Image](https://github.com/user-attachments/assets/26b4b06d-e58e-4725-be17-2358511dc424)

You could also try using Paletta tool to check, compare or find colors with required contrast ratio.
https://lab.highsoft.com/paletta/

**Comment by TorsteinHonsi** [comment]
@jperals You've touched upon a dilemma with the default colors. While we want sufficient contrast against the background, we also want sufficient contrast _between_ the colors. Unfortunately we can't have it both ways, so the default theme is a tradeoff. We try to achieve acceptable contrast between any neighboring colors, otherwise for example a pie chart would suffer.

We have an [accessible theme](https://www.highcharts.com/demo/highcharts/accessible-line) where the series colors have better against the background, at the cost of comparing lines to the legend. In this case saved by dash style.

As you suggest, in this scenario I would consider different strategies for preserving the contrast of the error bars

> One way to mitigate this would be to let the error bar marker have an outline or a semitransparent overlay

You could achieve that using a shim error bar behind it: https://jsfiddle.net/highcharts/qa24dxt7/

Other strategies could be changing the fill colors or the columns, or [moving the error bars to the side of the column](https://jsfiddle.net/highcharts/qa24dxt7/1/)

**Comment by jperals** [comment]
Thanks for the suggestions. The shim error bar workaround would not really work for us as our implementation unfortunately does not work well with styled mode.

Would it be a possibility for us to contribute the optional outlines as a feature at a later point?

**Comment by KacperMadej** [comment]
You have submitted the demo using styled mode in the top comment. Feel free to update your comments with new, more, or relevant demos to better show the problem you are having.

Here's a different way of doing the same - this time no styled mode: https://jsfiddle.net/BlackLabel/xL7snu0q/

> Would it be a possibility for us to contribute the optional outlines as a feature at a later point?

Yes, you can find more info here: https://github.com/highcharts/highcharts/blob/master/CONTRIBUTING.md
If anything is unclear please let us know.
